### ISSUE TITLE
Add directory as non-root directory via put file

### ISSUE THREAD

**Issue by hobofan** [pachyderm/pachyderm#5249]
**What happened?**:

I tried to add a directory full of files to a non-root directory, and it fails as a non-descriptive error:
```
$ pachctl put file -r images@master:/training/images/ -f "Training - Images"
attempted to visit "/training/images/20191018_S2_1-11.tif", but it's not a directory
```

The specified directory is created, but the files are not added.

Adding the same files to the root directory (`pachctl put file -r images@master:/ -f "Training - Images"`) works as expected.

**What you expected to happen?**:

Either add the files, as I expected it to work, or if I am somehow misinterpreting how it should be used, a better error message would be good.

**How to reproduce it (as minimally and precisely as possible)?**:

```
pachctl create repo images
mkdir foo
touch foo/toasty.txt
pachctl put file -r images@master:/training/images/ -f foo/
```

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T21:52:18Z", GoVersion:"go1.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:18:29Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Pachyderm CLI and pachd server version (use `pachctl version`):
```
COMPONENT           VERSION
pachctl             1.10.5
pachd               1.10.5
```
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):

local Kubernetes via macOS Docker Desktop

- OS (e.g. from /etc/os-release):

macOS Mojave 10.14.6 (18G103)

- Others:

**Comment by JoeyZwicker** [comment]
Thanks @hobofan and sorry you're hitting this. We'll play with the repro steps and figure out what's going on. I've also experienced some finickiness with `put file -r`.

**Comment by pappasilenus** [comment]
Hi, @hobofan, I ran tests on both 1.11.3 and 1.10.5 and I could not reproduce this.  I thought, perhaps, we weren't creating intermediate directories (like `mkdir -p` does), but it looks like we are.

I tried with a source directory with spaces in the name, and that works properly.  I tried it with a destination path in the repo that has spaces in the name, and that works properly.

Can you paste attached appropriately sanitized `pachctl logs` that correspond to the execution of this command?

**Comment by hobofan** [comment]
Thanks for looking into it @pappasilenus!

I tried to replicate the issue with both the reproduction snippet I posted and the directory with spaces as well, but no success. When the issue appeared I had just freshly updated Kubernetes from and older setup and Pachyderm for the first time, with some startup complications in the beginning (which I also didn't have since). So I guess it might have just been some wonky transient effects from all of that 🤷 , and close the issue!
